dashboard/app/assets/javascripts/progress.js

```diff
@@ -76,12 +76,15 @@ window.dashboard.progress = (function () {
     }
   };
 
-  progress.renderStageProgress = function (stageData, progressData, clientProgress, currentLevelId) {
+  progress.renderStageProgress = function (stageData, progressData, clientProgress, currentLevelId, page) {
     var serverProgress = progressData.levels || {};
     var currentLevelIndex = null;
 
     var combinedProgress = stageData.levels.map(function(level, index) {
-      if (level.id === currentLevelId) {
+      // Determine the current level index.
+      // However, because long assessments can have the same level appearing
+      // multiple times, just set this the first time it's determined.
+      if (level.id === currentLevelId && currentLevelIndex === null) {
```

Comment:
hmm.. I wonder if we can use the url instead of the level id to determine what the current level is.